RecoHI/HiTracking/plugins/HIPixelTracksSlimmer_miniAOD.cc

```diff
@@ -0,0 +1,256 @@
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+  const edm::EDGetTokenT<reco::TrackCollection> srcTracks_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcVertices_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcPrimaryVertices_;
+  const double dzSigCut_;
+  const double dxySigCut_;
+  const double dzSigHP_;
+  const double dxySigHP_;
+  const double ptMax_;
+  const int covarianceVersion_;
+  const int covarianceSchema_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HIPixelTracksSlimmer_miniAOD::HIPixelTracksSlimmer_miniAOD(const edm::ParameterSet& iConfig)
+    : srcTracks_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("srcTracks"))),
+      srcVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcVertices"))),
+      srcPrimaryVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcPrimaryVertices"))),
+      dzSigCut_(iConfig.getParameter<double>("dzSigCut")),
+      dxySigCut_(iConfig.getParameter<double>("dxySigCut")),
+      dzSigHP_(iConfig.getParameter<double>("dzSigHP")),
+      dxySigHP_(iConfig.getParameter<double>("dxySigHP")),
+      ptMax_(iConfig.getParameter<double>("ptMax")),
+      covarianceVersion_(iConfig.getParameter<int>("covarianceVersion")),
+      covarianceSchema_(iConfig.getParameter<int>("covarianceSchema")) {
+  produces<std::vector<reco::Track>>();
+  produces<std::vector<pat::PackedCandidate>>();
+  produces<edm::Association<pat::PackedCandidateCollection>>();
+}
+
+HIPixelTracksSlimmer_miniAOD::~HIPixelTracksSlimmer_miniAOD() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void HIPixelTracksSlimmer_miniAOD::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace std;
+
+  //track collection
+  edm::Handle<reco::TrackCollection> tracks;
+  iEvent.getByToken(srcTracks_, tracks);
+
+  auto outPtrTrks = std::make_unique<std::vector<reco::Track>>();
+  auto outPtrTrksAsCands = std::make_unique<std::vector<pat::PackedCandidate>>();
+
+  //vtx collections
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(srcVertices_, vertices);
+
+  edm::Handle<reco::VertexCollection> pvs;
+  iEvent.getByToken(srcPrimaryVertices_, pvs);
+  reco::VertexRef pv(pvs.id());
+  reco::VertexRefProd pvRefProd(pvs);
+  if (!pvs->empty()) {
+    pv = reco::VertexRef(pvs, 0);
+  }
+
+  //best vertex
+  double bestvz = -999.9, bestvx = -999.9, bestvy = -999.9;
+  double bestvzError = -999.9, bestvxError = -999.9, bestvyError = -999.9;
+  const reco::Vertex& vtx = (*vertices)[0];
```

Comment:
is use of the original pre-slimming vertex here justified?
The "pvs" have all parameters used here already, except for the smaller precision on the covariance/uncertainty values (rounded to 10 bits, as seen in PATVertexSlimmer).
I propose to use ```pvs``` directly and drop this.

Reply:
Okay, I now we use only after slimming vertex. Please, see this commit: https://github.com/cms-sw/cmssw/pull/30313/commits/8a0f07e9c2f43e4a3fec4340121025a5a1549ac4